### PR TITLE
rgw: [CORTX-30300] Added a sample IAM policy unittest

### PR DESCRIPTION
**Pull request by vastradparayya**
Problem:
	Explore and add a sample IAM policy unittest
Solution:
	Explored and added a sample IAM policy unittest

Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/test/rgw/test_rgw_iam_policy.cc [hunk 1]
```diff
@@ -732,6 +732,20 @@ TEST_F(PolicyTest, Eval7) {
 	    Effect::Pass);
 }
 
+TEST_F(PolicyTest, eval_conditions) {
+  auto p  = Policy(cct.get(), arbitrary_tenant,
+		   bufferlist::static_from_string(example1));
+  Environment e;
+
+  // evaluates statements, default allowed will be false
+  EXPECT_EQ(p.eval_conditions(e),
```
Comment (vastradparayya) [inline src/test/rgw/test_rgw_iam_policy.cc:741]: eval_conditions function from rgw_iam_policy.cc not covered in existing unittest. So covering `eval_conditions` as sample testcase

**Review by sachinpunadikar — APPROVED** [review]
+1

**Comment by shailesh-vaidya** [comment]
retest this please

**Comment by vastradparayya** [comment]
Closing this PR as this is covered in #316 #318 and #323

**Pull request closed without merging**